python/cudf/cudf/_lib/pylibcudf/strings/find.pyx

```diff
@@ -0,0 +1,255 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+from libcpp.memory cimport unique_ptr
+from libcpp.utility cimport move
+
+cimport cudf._lib.cpp.strings.find as cpp_find
+from cudf._lib.cpp.column.column cimport column
+from cudf._lib.pylibcudf.column cimport Column
+from cudf._lib.pylibcudf.scalar cimport Scalar
+
+from cython.operator import dereference
+
+from cudf._lib.cpp.scalar.scalar cimport string_scalar
+
+
+cpdef Column find(
+    Column input,
+    ColumnOrScalar target,
+    size_type start=0,
+    size_type stop=-1
+):
+    """Perform a string find operation.
```

Comment:
We should use a more descriptive docstring here. Something like "Find the first position of target in each input" or something.